source/rust_verify/src/rust_to_vir_expr.rs

```diff
@@ -936,6 +948,83 @@ fn fn_call_to_vir<'tcx>(
         Box::new(std::iter::repeat(None))
     };
 
+    if is_new_strlit {
+        let s = if let ExprKind::Lit(lit0) = &args[0].kind {
+            if let rustc_ast::LitKind::Str(s, _) = lit0.node {
+                s
+            } else {
+                panic!("unexpected arguments to new_strlit")
+            }
+        } else {
+            panic!("unexpected arguments to new_strlit")
+        };
+
+        let c = vir::ast::Constant::StrSlice(Arc::new(s.to_string()));
+        return Ok(mk_expr(ExprX::Const(c)));
+    }
+
+    if is_reveal_strlit {
+        let s = if let ExprKind::Lit(lit0) = &args[0].kind {
+            if let rustc_ast::LitKind::Str(s, _) = lit0.node {
+                s
+            } else {
+                panic!("unexpected arguments to new_strlit")
+            }
+        } else {
+            panic!("unexpected arguments to new_strlit")
+        };
+
+        return Ok(mk_expr(ExprX::RevealString(Arc::new(s.to_string()))));
+    }
+
+    if is_strslice_get_char {
+        return match &expr.kind {
+            ExprKind::Call(_, args) => {
+                assert!(args.len() == 2);
+                let arg0 = args.first().unwrap();
+                let arg0 = expr_to_vir(bctx, arg0, ExprModifier::REGULAR)
+                    .expect("internal compiler error");
```

Comment:
Please briefly describe what's unexpected here, instead. "invalid parameter for strslice_get_char", for example.

Reply:
Done